test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/awsquerycompatible/AwsQueryCompatibleErrorTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.services.awsquerycompatible;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.anyUrl;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import java.net.URI;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.querycompatiblejson.QueryCompatibleJsonAsyncClient;
+import software.amazon.awssdk.services.querycompatiblejson.QueryCompatibleJsonClient;
+import software.amazon.awssdk.services.querycompatiblejson.model.AllTypeResponse;
+import software.amazon.awssdk.services.querycompatiblejson.model.QueryCompatibleJsonResponse;
+import software.amazon.awssdk.utils.builder.SdkBuilder;
+
+public class AwsQueryCompatibleErrorTest {
```

Comment:
Can we add test cases with different queryHeaderValue like one with some  spaces before and after the value , you had mentioned we no longer need trimming.
Can we add some cases where queryHeaderValue can have unexpected values

Reply:
So the value of these tests I added here is to mock end-to-end codegen logic for the new `AwsQueryCompatible` trait. As the codegen doesn't control or verify the validity of the header value, I don't see any value or know how to implement the requested test scenarios. The requested test scenarios were already addressed in the `aws-json-protocol` parameterized tests. Let me know if you have a different opinion.